Backend/app/rider-platform/rider-app/Main/src/SharedLogic/Confirm.hs

```diff
@@ -176,31 +172,58 @@ confirm DConfirmReq {..} = do
       when (driverOffer.validTill < now) $ throwError $ QuoteExpired quote.id.getId
       pure driverOffer.bppQuoteId
 
-    checkOverlap :: (ServiceFlow m r) => Int -> Int -> UTCTime -> Maybe DL.Location -> DM.Merchant -> DRB.Booking -> m Bool
-    checkOverlap estimatedDistance estimatedDuration curBookingStartTime currBookingToLocation merchant booking = do
-      destToScheduledPickup <- calculateDistanceToScheduledPickup currBookingToLocation booking
-      let estimatedDistanceInKm' = destToScheduledPickup + estimatedDistance
-          estimatedDistanceInKm = estimatedDistanceInKm' `div` 1000
-          estRideEndTimeByDuration = addUTCTime (intToNominalDiffTime estimatedDuration + merchant.scheduleRideBufferTime) curBookingStartTime
-          estRideEndTimeByDist = addUTCTime (intToNominalDiffTime (estimatedDistanceInKm * 3 * 60) + merchant.scheduleRideBufferTime) curBookingStartTime -- TODO: need to make avg speed at rider side configurable : current 3min/km
-      return $ max estRideEndTimeByDuration estRideEndTimeByDist >= booking.startTime
+checkOverlap :: UTCTime -> UTCTime -> DSReq.RiderPreferredOption -> DM.Merchant -> DRB.Booking -> Bool
+checkOverlap curBookingStartTime currBookingEndTime searchType merchant scheduledBooking =
+  let bookingEndTime = calcBookingEndTime merchant scheduledBooking
+   in case searchType of
+        DSReq.Ambulance -> False
+        _ -> isRidePossible scheduledBooking.startTime bookingEndTime curBookingStartTime currBookingEndTime
 
-    calculateDistanceToScheduledPickup :: (ServiceFlow m r) => Maybe DL.Location -> DRB.Booking -> m Int
-    calculateDistanceToScheduledPickup currBookingToLocation booking =
-      case currBookingToLocation of
-        Nothing -> return 0
-        Just loc -> do
-          let currBookingDest = LatLong {lat = loc.lat, lon = loc.lon}
-          let scheduledPickup = LatLong {lat = booking.fromLocation.lat, lon = booking.fromLocation.lon}
-          distance <- do
-            Maps.getDistanceForScheduledRides booking.merchantId booking.merchantOperatingCityId $
-              Maps.GetDistanceReq
-                { origin = currBookingDest,
-                  destination = scheduledPickup,
-                  travelMode = Just Maps.CAR,
-                  distanceUnit = Meter
-                }
-          return $ distance.distance.getMeters
+isRidePossible :: UTCTime -> UTCTime -> UTCTime -> UTCTime -> Bool
+isRidePossible bookingStartTime bookingEndTime searchStartTime searchEndTime =
+  (bookingStartTime <= searchStartTime && searchStartTime <= bookingEndTime)
+    || (bookingStartTime <= searchEndTime && searchEndTime <= bookingEndTime)
```

Comment:
`searchStartTime > bookingEndTime || searchEndTime < bookingStartTime` - Only this should be enough?